.github/workflows/ci-tests.yml

```diff
@@ -54,17 +51,18 @@ jobs:
         uses: KyleMayes/install-llvm-action@8852e4d5c58653ed05135c0a5d949d9c2febcb00 # v1.6.1
         with:
           version: "15.0"
-        if: runner.os == 'macOS'
+        if: contains(fromJSON('["macOS", "macos-12", "Windows", "windows-latest", "windows-2022"]'), runner.os)
 
       - name: Install Protoc
-        uses: arduino/setup-protoc@v1
+        uses: arduino/setup-protoc@9b1ee5b22b0a3f1feb8c2ff99b32c89b3c3191e9 # v2.0.0
         with:
           repo-token: ${{ secrets.GITHUB_TOKEN }}
+        if: contains(fromJSON('["Linux", "ubuntu-20.04", "macOS", "macos-12", "windows-latest", "windows-2022"]'), runner.os)
 
       # Workaround to resolve link error with C:\msys64\mingw64\bin\libclang.dll
       - name: Remove msys64
         run: Remove-Item -LiteralPath "C:\msys64\" -Force -Recurse
-        if: runner.os == 'Windows'
+        if: contains(fromJSON('["windows-latest", "windows-2022"]'), runner.os)
```

Comment:
`runner.os` is ALWAYS exactly `Windows` and nothing else for all versions for GitHub's runners. You're confusing this again with `matrix.os`, this change breaks support for GitHub's runners. Just add this:
```
        # May not exist on self-hosted runners
        continue-on-error: true
```